transmart-rest-api/src/func-test/groovy/org/transmartproject/rest/v2/ApiAuditInterceptorSpec.groovy

```diff
@@ -0,0 +1,358 @@
+package org.transmartproject.rest.v2
+
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.http.HttpStatus
+import org.transmartproject.core.log.AccessLogEntry
+import org.transmartproject.core.log.AccessLogEntryResource
+import org.transmartproject.interceptors.ApiAuditInterceptor
+import org.transmartproject.mock.MockAccessLogEntryResource
+import org.transmartproject.rest.BusinessExceptionController
+import spock.lang.Unroll
+
+import static org.springframework.http.HttpMethod.*
+
+@Slf4j
+class ApiAuditInterceptorSpec extends V2ResourceSpec {
```

Comment:
The main reason to switch from unit test to integration test is that the unit testing framework has difficulty to handle mandatory autowired dependencies. It would be nice if the grails testing framework offers easy mocking of injected beans, like the comination of JUnit, Spring testing and Mockito does.
It now checks if something is written to the log service, instead of matching of calls. It looks like the interceptor causes asynchronous behaviour, whithout the `sleep(1000)` I got errors. I will try if they are still needed.

Reply:
Removing the `sleep` still makes them all fail. Weird, because you are supposed to be able to alter the response in the `after()` function. I would then expect that the action has completed before sending the response.